include/util/dist_table_wrapper.hpp

```diff
@@ -46,6 +46,30 @@ template <typename T> class DistTableWrapper
         return table_[index];
     }
 
+    void InvalidateRoute(NodeID from, NodeID to)
+    {
+        BOOST_ASSERT_MSG(from < number_of_nodes_, "from ID is out of bound");
+        BOOST_ASSERT_MSG(to < number_of_nodes_, "to ID is out of bound");
+
+        const auto index = from * number_of_nodes_ + to;
+
+        BOOST_ASSERT_MSG(index < table_.size(), "index is out of bound");
+
+        table_[index] = INVALID_EDGE_WEIGHT;
+    }
+
+    void ShortcutRoute(NodeID from, NodeID to)
```

Comment:
I'm not totally convinced that `ShortcutRoute ` and `InvalidateRoute ` are good names. If you want to keep them more general as they are now, can you add a comment why we do this here? so something like
`force the roundtrip to start at source and end at destination by virtually squashing them together`

or you can maybe name them `squashSourceAndDestination` and `separateDestination`